en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,179 @@
+However at time of writing, only XChaCha20 can actually be used (AES can be selected, but there is no implementation).
+:::
+
+## How ULog Encryption Works
+
+When the ULog file is first opened:
+
+- If [SDLOG_ALGORITHM](../advanced_config/parameter_reference.md#SDLOG_ALGORITHM) is set to default, a symmetric XChaCha20 key is generated and wrapped (encrypted) with RSA2048 public key encryption
+- This wrapped key is stored on the SD Card with the ".ulgk" suffix.
+
+While the ULog file is written to disk:
+
+- The generated (plaintext) key is used to encrypt the data blocks right before writing to disk.
+
+After the flight, there are two files:
+
+- `.ulogc` (ulog cipher): the actual log file data, encrypted
+- `.ulogk` (ulog wrapped key): the symmetric key, wrapped with rsa
+
+However, these files will both have the normal `.ulg` suffix if downloading from QGroundControl.
+
+## Creating a Flight Controller Build that contains Log Encryption
+
+Crypto uses large amounts of flash memory, and is therefore not included in PX4 builds by default (the exception is the make target `px4-fmu-v5_cryptotest`).
+Below we show how to add log encryption functionality to a flight controller target, using the `px4-fmu-v5` board as an example.
+
+Reference the `cryptotest.px4board` file in `boards/px4/fmu-v5` for the arguments you will need to add to the `.px4board` file associated with the flight controller you are targeting.
+You could also copy and paste `cryptotest.px4board` to the other flight controller's `boards/` location if you want this to be a separate build entirely (in that case, your `make` command would include `_cryptotest` at the end to add log encryption).
+
+### Crypto .px4board arguments
+
+| Argument                     | Description                                                                                                                        |
+| ---------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| CONFIG_BOARD_CRYPTO          | Requirement for log encryption functionality to be enabled.<br />= `y`: Enable log encryption.<br />= `n`: Disable log encryption. |
+| CONFIG_DRIVERS_SW_CRYPTO     | Enables the built-in PX4 crypto backend library to handle logfile encryption.<br />= `y`: Enable<br />= `n`: Disable               |
+| CONFIG_DRIVERS_STUB_KEYSTORE | Enables the built-in PX4 stub keystore driver.<br />= `y`: Enable<br />= `n`: Disable                                              |
+| CONFIG_PUBLIC_KEY0           | Location of public key for keystore index 0.<br />= `{path to key0}`                                                               |
+| CONFIG_PUBLIC_KEY1           | Location of public key for keystore index 1.<br />= `{path to key1}`                                                               |
```

Comment:
Does this value define default keys that you can use, and if so, where can you find the corresponding private key for testing?

1. Where should these files be stored? Is there a standard location.
2. Is this everything you need to do to fetch the file and include it in firmware?
3. What is the difference between the two keystores? Do you need both? Which is used?